data-prepper-core/src/main/java/org/opensearch/dataprepper/core/pipeline/server/UpdatePipelineBaseHandler.java

```diff
@@ -0,0 +1,175 @@
+
+package org.opensearch.dataprepper.core.pipeline.server;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.sun.net.httpserver.HttpExchange;
+import org.opensearch.dataprepper.core.pipeline.PipelinesProvider;
+import org.opensearch.dataprepper.model.configuration.PipelinesDataFlowModel;
+import org.opensearch.dataprepper.pipeline.parser.PipelineConfigurationS3FileReader;
+import org.opensearch.dataprepper.pipeline.parser.PipelinesDataflowModelParser;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import software.amazon.awssdk.core.exception.SdkClientException;
+import software.amazon.awssdk.regions.Region;
+
+import javax.ws.rs.HttpMethod;
+import java.io.BufferedReader;
+import java.io.IOException;
+import java.io.InputStreamReader;
+import java.net.HttpURLConnection;
+import java.nio.charset.StandardCharsets;
+import java.util.List;
+import java.util.Map;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+import java.util.stream.Collectors;
+
+/**
+ * HttpHandler to handle requests for updating pipeline configurations from S3
+ */
+public abstract class UpdatePipelineBaseHandler {
+
+    private static final Logger LOG = LoggerFactory.getLogger(UpdatePipelineBaseHandler.class);
+    private static final ObjectMapper OBJECT_MAPPER = new ObjectMapper();
+    private static final Pattern PIPELINE_NAME_PATTERN = Pattern.compile("/([a-zA-Z0-9-]{1,28})$");
+    private final PipelinesProvider pipelinesProvider;
+
+    public UpdatePipelineBaseHandler(final PipelinesProvider pipelinesProvider) {
+        this.pipelinesProvider = pipelinesProvider;
+    }
+
+    public void baseHandle(final HttpExchange exchange, boolean executeUpdate) throws IOException {
+        final String requestMethod = exchange.getRequestMethod();
+        if (!requestMethod.equals(HttpMethod.PUT)) {
+            exchange.sendResponseHeaders(HttpURLConnection.HTTP_BAD_METHOD, 0);
+            exchange.getResponseBody().close();
+            return;
+        }
+
+        try {
+            // Extract pipeline name from URL path
+            final String pipelineName = extractPipelineNameFromPath(exchange.getRequestURI().getPath());
+            if (pipelineName == null) {
+                sendErrorResponse(exchange, HttpURLConnection.HTTP_BAD_REQUEST, "Invalid pipeline name in URL path");
+                return;
+            }
+
+            // Parse request body to get S3 path
+            final String requestBody = new BufferedReader(
+                    new InputStreamReader(exchange.getRequestBody(), StandardCharsets.UTF_8))
+                    .lines()
+                    .collect(Collectors.joining("\n"));
+
+            final S3PathRequest s3PathRequest = parseS3PathRequest(requestBody);
+            PipelinesDataFlowModel targetPipelinesDataFlowModel =
+                    new PipelinesDataflowModelParser(
+                            new PipelineConfigurationS3FileReader(s3PathRequest.s3paths, s3PathRequest.s3region)
```

Comment:
We should support reloading from the local file system as well. This will help with local testing and the additional effort should be light. It just requires a little code clean-up to determine the file path. e.g. `s3paths` versus `filepaths`.